test/lsp/lsp_rename.ml

```diff
@@ -0,0 +1,191 @@
+  let server, prog = add_cobol_doc server ~projdir "prog.cob" doc in
+  let rename_at_position ?key (position: Position.t) =
+    let params = RenameParams.create ~newName:"aNewName" ~position ~textDocument:prog () in
+    Pretty.out "%a(line %d, character %d):\n"
+      Fmt.(option ~none:nop (string ++ sp)) key
+      position.line position.character;
+    begin
+      try
+      match LSP.Request.rename server params with
+      | Error e ->
+        Pretty.out "Renamed failed: %S@." e
+      | Ok { changes = None; _ } ->
+        Pretty.out "No renames@."
+      | Ok { changes = Some assoc; _ } ->
+        Pretty.out "@.@[<hv 4>%d rename entries:@;%a@]@\n"
+          (count assoc)
+          (Fmt.list ~sep:Fmt.sp pp_assoc_elem) assoc
+      with _ -> Pretty.out "Failed rename@."
+    end
+  in
+  StringMap.iter (fun n p -> rename_at_position ~key:n p) positions.pos_map;
+  List.iter (fun p -> rename_at_position p) positions.pos_anonymous;
+  end_with_postproc
+
+let%expect_test "rename" =
+  let end_with_postproc = rename_positions @@ extract_position_markers {cobol|
+        IDENTIFICATION DIVISION.
+        PROGRAM-ID. prog.
+        DATA DIVISION.
+        WORKING-STORAGE SECTION.
+        01 o_|_ld-name PIC 9.
+        PROCEDURE DIVISION.
+          MOVE 1 TO old-na_|_me.
+          S_|_TOP RUN.
+  |cobol}
+  in
+  end_with_postproc [%expect.output];
+  [%expect {|
+    {"params":{"diagnostics":[],"uri":"file://__rootdir__/prog.cob"},"method":"textDocument/publishDiagnostics","jsonrpc":"2.0"}
+    (line 5, character 12):
+    2 rename entries:
+        aNewName at __rootdir__/prog.cob:8.20-8.28:
+           5           WORKING-STORAGE SECTION.
+       6           01 old-name PIC 9.
+       7           PROCEDURE DIVISION.
+       8 >           MOVE 1 TO old-name.
+    ----                       ^^^^^^^^
+       9             STOP RUN.
+      10
+    aNewName at __rootdir__/prog.cob:6.11-6.19:
+       3           PROGRAM-ID. prog.
+       4           DATA DIVISION.
+       5           WORKING-STORAGE SECTION.
+       6 >         01 old-name PIC 9.
+    ----              ^^^^^^^^
+       7           PROCEDURE DIVISION.
+       8             MOVE 1 TO old-name.
+    (line 7, character 26):
+    2 rename entries:
+        aNewName at __rootdir__/prog.cob:8.20-8.28:
+           5           WORKING-STORAGE SECTION.
+       6           01 old-name PIC 9.
+       7           PROCEDURE DIVISION.
+       8 >           MOVE 1 TO old-name.
+    ----                       ^^^^^^^^
+       9             STOP RUN.
+      10
+    aNewName at __rootdir__/prog.cob:6.11-6.19:
+       3           PROGRAM-ID. prog.
+       4           DATA DIVISION.
+       5           WORKING-STORAGE SECTION.
+       6 >         01 old-name PIC 9.
+    ----              ^^^^^^^^
+       7           PROCEDURE DIVISION.
+       8             MOVE 1 TO old-name.
+    (line 8, character 11):
+    0 rename entries: |}]
+
+let%expect_test "rename-copybook" =
```

Comment:
```suggestion
let%expect_test "rename-with-a-ref-in-a-copybook" =
```

Reply:
On the VSCode side, what happens in this case? Could we just proceed with the renaming anyways, and raise a warning instead while doing so? (Could be a behavior that is conditional on a config flag given to `Lsp_request.rename`).

For the future (maybe a new issue): Ideally that'd be something interactive with a yes/no answer, but I don't know if the protocol offers enough tools for this (there's probably a way to intercept requests on the extension's side).

Reply:
![image](https://github.com/user-attachments/assets/70feb606-0010-4f08-8096-04eb0c0d6792)

Atm the rename is cancelled and this message appears